### PR TITLE
refactor(js-loading): check for a data-attribute before invoking a mo…

### PR DESCRIPTION
**Pull request by Wandkleister**
…dule

### CODE REVIEW

File: src/_includes/js/locations.js [hunk 1]
```diff
@@ -1,12 +1,14 @@
 export class Locations {
 	constructor() {
 		this.container = document.querySelector(".locations");
-		this.tablist = this.container.querySelector("[role=tablist]");
-		this.tabs = this.container.querySelectorAll("[role=tab]");
-		this.tabpanels = this.container.querySelectorAll("[role=tabpanel]");
-		this.activeTab = this.container.querySelector("[role=tab][aria-selected=true]");
 
-		this.bindEvents();
+		if (this.container) {
```
Comment (s2-schwaddi) [inline src/_includes/js/locations.js:5]: consider: `if (! ... || ....) return;`

File: src/_layouts/default.liquid [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <!DOCTYPE html>
 <html lang="en">
   {% include head %}
-  <body class="{{bodyClass}}">
+  <body class="{{bodyClass}}" data-js-item="{{dataJsItem}}">
```
Comment (s2-schwaddi) [inline src/_layouts/default.liquid:4]: Why does this one need to be dynamic?

Comment (Wandkleister) [reply src/_layouts/default.liquid:4]: The attribute here is used to load the background color transition of the page. As I'm unsure if we will have that transition on every page, I wanted to have it as an optional part in the template.
And also we could add another `dataJsItem` here, if there would be the need for another page-wide module.

File: src/_includes/js/animate-slidein.js [hunk 3]
```diff
@@ -2,6 +2,8 @@ export class SlideIn {
 	titles = document.querySelectorAll(".slide-in");
 	hideAnimations = window.matchMedia("(prefers-reduced-motion: reduce)");
 	constructor() {
+		if (!this.titles) return;
```
Comment (s2-schwaddi) [inline src/_includes/js/animate-slidein.js:5]: `if (!this.titles.length) return;`?

File: src/_includes/js/main.js [hunk 4]
```diff
@@ -8,12 +8,37 @@ import { SlideIn } from "./animate-slidein";
...
+
+	if (isModuleNeeded("offerings")) {
+		new Offerings();
+	}
+
+	if (isModuleNeeded("slide-in")) {
+		new SlideIn();
+	}
+
 	new Parallax();
```
Comment (s2-schwaddi) [inline src/_includes/js/main.js:39]: this isn't initialized dynamically?

File: src/_includes/js/animate-slidein.js (hunk 3 above)
Comment (Wandkleister) [reply src/_includes/js/animate-slidein.js]: Changed to `if (!this.titles.length) return;`

File: src/_includes/js/main.js (hunk 4 above)
Comment (Wandkleister) [reply src/_includes/js/main.js:39]: I was thinking if I'm able to use the `isModuleNeeded` pattern here. But then when looking into the `parallax.js` we have 4 conditionals to check for domElements or prefers-reduced-motion. 

Do you think its better to have something like this?
`	
if (isModuleNeeded("parallax-fast") || isModuleNeeded("parallax-medium") || isModuleNeeded("parallax-slow")) {
		new Parallax();
	}
`

Comment (s2-schwaddi) [reply src/_includes/js/main.js:39]: If the pattern isn't applicable here please make sure that it doesn't throw errors.

Comment (Wandkleister) [reply src/_includes/js/main.js:39]: I adjusted the parallax.js file to check for the length of the NodeLists

**Review by s2-schwaddi — APPROVED** [review]

**Pull request merged**